test/common/vmimages.py

```diff
@@ -114,9 +118,74 @@ def enough_disk_space():
     free = st.f_bavail * st.f_frsize / (1024*1024*1024)
     return free >= PRUNE_THRESHOLD_G;
 
+def get_refs():
+    """Return dictionary for available refs of the format {'rhel-7.4': 'ad50328990e44c22501bd5e454746d4b5e561b7c'}
+       Expects to be called from the top level of the git checkout
+    """
+    # get all remote heads and filter empty lines
+    # output of ls-remote has the format
+    #
+    # d864d3792db442e3de3d1811fa4bc371793a8f4f	refs/heads/master
+    # ad50328990e44c22501bd5e454746d4b5e561b7c	refs/heads/rhel-7.4
+    refs = filter(None, subprocess.check_output(["git", "ls-remote",  "--heads"]).split("\n"))
+    # filter out the "refs/heads/" prefix
+    # and generate a dictionary
+    prefix = "refs/heads"
+    return dict(map(lambda ref: [s[s.startswith(prefix) and len(prefix):] for s in reversed(ref.split())], refs))
+
+def get_image_links(ref):
+    """Return all image links for the given git ref
+       Expects to be called from the top level of the git checkout
+    """
+    # get all the links we have first
+    # trailing slash on GIT_IMAGE_PATH is important
+    git_path = GIT_IMAGE_PATH if GIT_IMAGE_PATH.endswith("/") else "{0}/".format(GIT_IMAGE_PATH)
+    try:
+        entries = filter(None, subprocess.check_output(["git", "ls-tree",  "--name-only", ref, git_path]).split("\n"))
+    except subprocess.CalledProcessError, e:
+        if e.returncode == 128:
+            sys.stderr.write("Skipping {0} due to tree error.\n".format(ref))
+            return []
+        raise
+    links = map(lambda entry: subprocess.check_output(["git", "show", "{0}:{1}".format(ref, entry)]), entries)
+    return filter(None, filter(lambda link: None if not link.endswith(".qcow2") else link, links))
```

Comment:
Matter of style of course, but IMHO using a list comprehension is more readable for such cases:
```python
    return [l for l in links if l.endswith('.qcow2')]
```

While I'm nitpicking, the first line could also be simplified, replacing the whole thing with
```python
for link in subprocess.check_output(["git", "show", "{0}:{1}".format(ref, entry)]).splitlines():
    if link.endswith('.qcow2'):
        yield link
```

This is a bit brittle in the sense that it's conceivable that there are files that contain lines ending with ".qcow"; but that's neither realistic nor would it be an actual problem, so let's keep it simple.